### DISCUSSION TITLE
User without permission to delete can see delete bulk action.

### DISCUSSION THREAD

**Discussion by otavio-araujo** [chiiya/filament-access-control#32] [Q&A]
``Users that does not have permission to delete still can see those checkboxes. When selected, and click on bulk button, it shows Delete bull button.

How can i solve that?

Anyone can help?

Changing the code of vendor's FilementUser, Role and PermissionsResource can solve this. But, how can i solve this without chenge vendor's files?

```
        ->bulkActions([
              BulkAction::make('delete')
              ->label(__('filament-access-control::default.actions.delete'))
                  ->action(fn (Collection $records) => $records->each->delete())
                  ->deselectRecordsAfterCompletion()
                  ->requiresConfirmation()
                  ->color('danger')
                  ->icon('heroicon-o-trash')
                  ->visible(fn () => Filament::auth()->user()->can('admin-users.delete')),
          ]);
```

The code above, solves this, but... o had to change inside vendor's files.

**Comment by chiiya** [comment]
`deleteAny()` should be the policy method for controlling bulk delete.

**Comment by otavio-araujo** [comment]
Thank you!
Foegot that... My bad...

Obter o Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: Elisha Witte ***@***.***>
Enviado: Monday, March 27, 2023 6:37:02 PM
Para: chiiya/filament-access-control ***@***.***>
Cc: Otávio Araújo ***@***.***>; Author ***@***.***>
Assunto: Re: [chiiya/filament-access-control] User without permission to delete can see delete bulk action. (Discussion #32)


deleteAny() should be the policy method for controlling bulk delete.
